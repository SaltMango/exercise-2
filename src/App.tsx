import React, { FC } from "react";
import { observer } from "mobx-react-lite";
import Counter from "./CounterStore";
import "./app.css";

const App = observer(({ count }: { count: Counter }) => {
  return (
    <div className="container">
      <div className="click">Clicks: {count.currentCount}</div>
      <div>
        <button onClick={count.incrementCount} className="button-style">
          Increment
        </button>
        <button onClick={count.decrementCount} className="button-style">
          Decrement
        </button>
      </div>
    </div>
  );
});

export default App;