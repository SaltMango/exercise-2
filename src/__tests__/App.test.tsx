import App from "../App";
import React from "react";
import TestRenderer from "react-test-renderer";
import Counter from "../CounterStore";

test("counter increment and decrement works correctly", () => {
  const count = new Counter();
  const tree = TestRenderer.create(<App count={count} />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe("Counter Store", () => {
  test("check initial count", () => {
    const counter = new Counter();
    expect(counter.currentCount).toBe(0);
  });
  test("count increment correctly", () => {
    const counter = new Counter();
    counter.incrementCount();
    expect(counter.currentCount).toBe(1);
  });
  test("count decrement correctly", () => {
    const counter = new Counter();
    counter.decrementCount();
    expect(counter.currentCount).toBe(-1);
  });
});
