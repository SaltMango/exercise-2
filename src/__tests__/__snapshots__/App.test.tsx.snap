// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter increment and decrement works correctly 1`] = `
<div
  className="container"
>
  <div
    className="click"
  >
    Clicks: 
    0
  </div>
  <div>
    <button
      className="button-style"
      onClick={[Function]}
    >
      Increment
    </button>
    <button
      className="button-style"
      onClick={[Function]}
    >
      Decrement
    </button>
  </div>
</div>
`;
